### PR TITLE
Parsing gene counts data from HDF5 file

### PR DESCRIPTION
**Pull request by robinpaul85**
## Description

1) Tentative implementation of parsing gene counts data for DE analysis from HDF5 files. 
2) Group names aded to header of DE window and in DE info table.
3) Parsing from text files is still available and should not affect other datasets that support DE.
## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: server/src/mds3.init.js [hunk 1]
```diff
@@ -1595,7 +1596,34 @@ async function validate_query_rnaseqGeneCount(ds, genome) {
 	also require that there's no duplicate samples in header line, so rust/r won't break
 	*/
 	{
-		const samples = (await getFirstLine(q.file)).trim().split('\t').slice(4)
+		let samples = []
+		if (!ds.queries.rnaseqGeneCount.storage_type || ds.queries.rnaseqGeneCount.storage_type == 'text') {
```
Comment (xzhou82) [inline server/src/mds3.init.js:1600]: shouldn't allow storage_type to be missing

Comment (robinpaul85) [reply server/src/mds3.init.js:1600]: The reason its optional, is that then we would have to change all other dataset files. Right now in the absence of `storage_type` in dataset file, its presumed that data is stored in a text file. In the future, we will convert all DE text files to HDF5.

Comment (xzhou82) [reply server/src/mds3.init.js:1600]: please pull sjpp branch and change storage_type into required property

File: server/routes/termdb.DE.ts [hunk 2]
```diff
@@ -41,17 +41,21 @@ function init({ genomes }) {
...
+                groups[]
+                        values[] // using integer sample id
+*/
 	if (param.samplelst?.groups?.length != 2) throw '.samplelst.groups.length!=2'
 	if (param.samplelst.groups[0].values?.length < 1) throw 'samplelst.groups[0].values.length<1'
 	if (param.samplelst.groups[1].values?.length < 1) throw 'samplelst.groups[1].values.length<1'
 	// txt file uses string sample name, must convert integer sample id to string
 	// txt file uses string sample name, must convert integer sample id to string
 
+	if (ds.queries.rnaseqGeneCount.storage_type) {
```
Comment (xzhou82) [inline server/routes/termdb.DE.ts:55]: no need for if here

File: shared/types/src/dataset.ts [hunk 3]
```diff
@@ -448,13 +448,16 @@ for a given region, the median signal from probes in the region is used to make
...
+                file: string
 }
 */
 
 type RnaseqGeneCount = {
+	/** Name of the HDF5 or text file */
 	file: string
 	samplesFile?: string
+	/** Storage_type for storing data. Will deprecate text files */
+	storage_type?: 'text' | 'HDF5'
```
Comment (xzhou82) [inline shared/types/src/dataset.ts:460]: remove `?` so it's required

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**